src/lib/utils/compiler.h

```diff
@@ -155,7 +155,27 @@
     #define BOTAN_DIAGNOSTIC_POP           	_Pragma("clang diagnostic pop")
   #endif
 
+#endif
 
+/*
+* Define Botan::unreachable()
+*
+* There is a pending WG21 proposal for `std::unreachable()`
+*   http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0627r3.pdf
+*/
+namespace Botan {
+
+[[noreturn]] BOTAN_FORCE_INLINE void unreachable()
```

Comment:
This has to either be a macro, or defined somewhere else - `compiler.h` is included into `build.h` which is included into `ffi.h`, which needs to be parseable by a C89 compiler.

[There should be a CI test for this]

Reply:
Fair point. I wasn't aware of this. Moved it to assert.h